client/client_update.go

```diff
@@ -84,39 +92,54 @@ func (u *UpdateClient) GetScheduledUpdate(api ApiRequester, server string,
 // URL.
 func (u *UpdateClient) getUpdateInfo(api ApiRequester, process RequestProcessingFunc,
 	server string, current *CurrentUpdate) (interface{}, error) {
-	postReq, getReq, err := makeUpdateCheckRequest(server, current)
+	reqs, err := makeUpdateCheckRequest(server, current)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to create update check request")
 	}
 
-	r, err := api.Do(postReq)
-	if err != nil {
-		log.Debugf("Failed sending device provides to the backend: Error: %v", err)
-		return nil, errors.Wrapf(err, "POST update check request failed")
-	}
-
-	defer r.Body.Close()
-
-	if r.StatusCode != http.StatusOK && r.StatusCode != http.StatusNoContent {
-
-		// Fall back to the GET (Open-Source) functionality on all error codes
-		if r.StatusCode >= 400 && r.StatusCode < 600 {
-
-			log.Debugf("device provides not accepted by the server. Response code: %d", r.StatusCode)
-
-			r, err = api.Do(getReq)
+	ok := false
+	var r *http.Response
+	for _, req := range reqs {
```

Comment:
Musing:
It's OK, I personally would have extended the tuple with a third request return parameter but.

It's also a little symptomatic that we start relying on go-to's :/

Reply:
How about: I'll put it inside a separate function instead. Then it should be easier to read, and I can replace the gotos with `return` (which is to exit `switch` and `for` simultaneously).

Reply:
Better now? I think I disagree with the tuples, since then you have to repeat the code three times, and can't do it list-style like I've done. But it's nice to get rid of the gotos, and it's also smaller and easier on the eyes now I think.

Reply:
much better :)